### DISCUSSION TITLE
A panic occurs when selecting Python suggestions, and Helix closes

### DISCUSSION THREAD

**Discussion by xteveg** [helix-editor/helix#10882] [Show and tell]
![a-panic-occurs-when-selecting-python-suggestions-and-helix-v0-oprakk38sf4d1](https://github.com/helix-editor/helix/assets/123645186/68b1f502-250e-4680-9b41-69589cd5c192)

Hi, i need help.

If I press enter on a suggestion using Python, like the one in the image, I get the following error and Helix stops working and returns to the terminal. What could it be?

Error -> panic

thread 'main' panicked at /home/steve/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ropey-1.6.1/src/slice.rs:656:9:
Attempt to slice past end of RopeSlice: slice end 2, RopeSlice length 1
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

My config:

```python
[language-server.pylsp]
command = "pylsp"

[language-server.pylsp.config.pylsp.plugins]
flake8 = {enabled = true, ignore = ["E501", "E305", "E302", "W293"]}
autopep8 = {enabled = false}
mccabe = {enabled = false}
pycodestyle = {enabled = false}
pyflakes = {enabled = false}
pylint = {enabled = false}
yapf = {enabled = false}

[[language]]
name = "python"
auto-format = false
language-servers = [{name = "pylsp"}]
indent = {tab-width = 4, unit = "    "}
shebangs = ["python"]```

**Comment by David-Else** [comment]
Please confirm which exact version of Helix you are running and consider changing this to a bug report rather than a discussion, adding triple ticks around the code makes it easier to read too, cheers :)

**Reply by xteveg** [reply]
version -> helix 24.3 (a8010441)

**Comment by the-mikedavis** [comment]
This looks like a duplicate of https://github.com/helix-editor/helix/issues/10689
